instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go

```diff
@@ -33,9 +34,15 @@ import (
 
 const (
 	tracerKey  = "otel-go-contrib-tracer"
+	skippedKey = "otel-go-contrib-skipped"
 	tracerName = "go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin"
 )
 
+// FilterFunc allows to skip tracing the incoming requests based on
+// request characteristics. Returns true means the filter rule is hit
+// and this request will not be recorded.
+type FilterFunc func(req *http.Request) bool
```

Comment:
Can we get this to align with the [`otelhttp.Filter`](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/common.go#L41) type?  Even if we don't reuse that type here, the signature is identical so having it be `true` to allow rather than reject would enable re-use of the numerous [pre-defined `Filter`s](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/filters/filters.go).

Reply:
updated